include/boost/geometry/policies/robustness/segment_ratio.hpp

```diff
@@ -254,16 +254,13 @@ public :
         return m_numerator > m_denominator;
     }
 
-    inline bool near_end() const
+    //! Returns a value between 0.0 and 1.0
+    //! 0.0 means: exactly in the middle
+    //! 1.0 means: exactly on one of the edges (or even over it)
```

Comment:
This replaces `near_end` (which had a threshold). This better allows more fluent behavior without thresholds.

Reply:
Why is `double` returned instead of coordinate type or even better distance/comparable_distance result type (since it's compared with distances/lengths)?
What if coordinate type and/or distance result type is more precise than double?
Isn't `fp_type` defined for this purpose (for now ignoring the fact that it's definition is wrong, see TODO)?

Reply:
Indeed, `fp_type` was defined for this purpose, thanks. Updated.